### PR TITLE
make dfs restore easier to read

### PR DESCRIPTION
**Pull request by smousa**
Changes:
Will keep good snapshots from backup, but destroys bad snapshots.

https://jira.zenoss.com/browse/CC-2411

### CODE REVIEW

File: dfs/restore.go [hunk 1]
```diff
@@ -30,295 +30,257 @@ var (
...
+			glog.Errorf("Could not read backup file: %s", err)
 			return err
 		}
+
 		switch {
-		case header.Name == BackupMetadataFile:
-			// Skip it, we've already got it
-		case strings.HasPrefix(header.Name, SnapshotsMetadataDir):
-			parts := strings.Split(header.Name, "/")
+		case hdr.Name == BackupMetadataFile:
```
Comment (iancmcc) [inline dfs/restore.go]: Why did you remove the helpful comment for helping

File: dfs/restore.go [hunk 2]
```diff
@@ -30,295 +30,257 @@ var (
...
-			cleanup()
+			glog.Errorf("Could not read backup file: %s", err)
 			return err
 		}
+
 		switch {
 		case hdr.Name == BackupMetadataFile:
-			// Skip it. We've already got it.
 		case strings.HasPrefix(hdr.Name, SnapshotsMetadataDir):
-			// This file is part of a volume snapshot. Find or create the pipe
```
Comment (iancmcc) [inline dfs/restore.go]: but comments

File: dfs/restore.go [hunk 3]
```diff
@@ -30,295 +30,269 @@ var (
...
-				w.Close()
-			}
-			if pw, ok := pipeStreams[k]; ok {
-				pw.Close()
-			}
-		}
+	return nil
+}
+
+// restoreV1 restores a backup created >= 1.1.3
```
Comment (iancmcc) [inline dfs/restore.go]: COMMENTS

File: dfs/restore.go [hunk 4]
```diff
@@ -30,295 +30,269 @@ var (
...
-			// This file is part of a volume snapshot. Find or create the pipe
+			// This file is part of a volume snapshot.  Find or create the pipe
 			// responsible for restoring that volume, strip off the extra
 			// parent path (so it resembles a tarfile containing the volume
 			// data at the root), and write the tar entry to the subpipe.
-			parts := strings.Split(hdr.Name, "/")
-			if len(parts) < 3 {
-				// This is a parent folder or something, not relevant
+			parts := strings.SplitN(hdr.Name, "/", 4)
+			if len(parts) <= 3 {
```
Comment (iancmcc) [inline dfs/restore.go]: no, why? <= is wrong, isn't it?

File: dfs/restore.go [hunk 5]
```diff
@@ -30,295 +30,269 @@ var (
...
-			// Find or create the pipe that's got a restoreVolume for this
+
+			id := path.Join(SnapshotsMetadataDir, tenant, label)
+			// Find or create the pipe that's got a restoreSnapshot for this
 			// volume reading from the other end
-			w := getSnapshotStream(tenant, label)
-			if w == nil {
+			s, ok := streamMap[id]
+			if !ok {
+				glog.Infof("Loading snapshot %s for tenant %s from backup", label, tenant)
```
Comment (iancmcc) [inline dfs/restore.go]: This should be debug

File: dfs/restore.go [hunk 6]
```diff
@@ -30,295 +30,269 @@ var (
...
-				startedDocker = true
+			parts := strings.SplitN(hdr.Name, "/", 2)
+			if len(parts) <= 1 {
+				continue
+			}
+
+			id := parts[0]
+			s, ok := streamMap[id]
+			if !ok {
+				glog.Infof("Loading docker images from backup")
```
Comment (iancmcc) [inline dfs/restore.go]: This should be debug

File: dfs/restore.go (hunk 4 above)
Comment (smousa) [inline dfs/restore.go]: if parts == 3 then there is nothing to pass to the tar
